cmds/core/cat/cat.go

```diff
@@ -25,36 +26,51 @@ var (
 	_ = flag.Bool("u", false, "ignored")
 )
 
-func catFile(w io.Writer, file string) error {
-	f, err := os.Open(file)
-	if err != nil {
-		return err
+func catFile(w io.Writer, files []*os.File) error {
+	for _, file := range files {
+		if _, err := io.Copy(w, file); err != nil {
+			return fmt.Errorf("error concatenating stdin to stdout: %v", err)
```

Comment:
This error message is not 100% correct because the reader might not necessarily be stdin.